nixos/modules/services/hardware/fw-fanctrl.nix

```
@@ -0,0 +1,129 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.hardware.fw-fanctrl;
+  defaultConfig = builtins.fromJSON (
+    builtins.readFile "${pkgs.fw-fanctrl}/share/fw-fanctrl/config.json"
+  );
+in
+{
+  options.hardware.fw-fanctrl = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Enable fw-fanctrl systemd service and install the needed packages.
+      '';
+    };
+    disableBatteryTempCheck = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Disable checking battery temperature sensor
+      '';
+    };
+    config = {
+      defaultStrategy = lib.mkOption {
+        type = lib.types.str;
+        default = defaultConfig.defaultStrategy;
+        description = "Default strategy to use";
+      };
+      strategyOnDischarging = lib.mkOption {
+        type = lib.types.str;
+        default = defaultConfig.strategyOnDischarging;
+        description = "Default strategy on discharging";
+      };
+      strategies = lib.mkOption {
+        default = defaultConfig.strategies;
+        type = lib.types.attrsOf (
+          lib.types.submodule (
+            { options, name, ... }:
+            {
+              options = {
+                fanSpeedUpdateFrequency = lib.mkOption {
+                  type = lib.types.int;
+                  default = 5;
+                  description = "How often the fan speed should be updated in seconds";
+                };
+                movingAverageInterval = lib.mkOption {
+                  type = lib.types.int;
+                  default = 25;
+                  description = "Interval (seconds) of the last temperatures to use to calculate the average temperature";
+                };
+                speedCurve = lib.mkOption {
+                  default = [ ];
+                  description = "How should the speed curve look like";
+                  type = lib.types.listOf (
+                    lib.types.submodule (
+                      { options, ... }:
+                      {
+                        options = {
+                          temp = lib.mkOption {
+                            type = lib.types.int;
+                            default = 0;
+                            description = "Temperature at which the fan speed should be changed";
+                          };
+                          speed = lib.mkOption {
+                            type = lib.types.int;
+                            default = 0;
+                            description = "Percent how fast the fan should run at";
+                          };
+                        };
+                      }
+                    )
+                  );
+                };
+              };
+            }
+          )
+        );
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    # Install package
```

```suggestion
    # Install packages
```

but that line is self explanatory

```suggestion
```